### PR TITLE
Format, lint, and check all sources

### PR DESCRIPTION
**Pull request by akutz**
**What this PR does / why we need it**:
This patch adds a target "check" that handles formatting, linting, and vetting project sources.

Related to https://github.com/kubernetes/test-infra/pull/13363.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:
Three new pre-submits will be created to ensure that all PRs are formatted, linted, and vetted before they're allowed to merge.

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
* The new "check" target in the Makefile ensures sources are formatted, linted, and vetted.
```

### CODE REVIEW

**Comment by akutz** [comment]
/assign @codenrhoden 
/assign @figo 
/assign @frapposelli 

Please lgtm. Thanks!

File: hack/shell-with-gopath.sh [hunk 1]
```diff
@@ -17,6 +17,7 @@
 # If SHELL_WITH_GOPATH is 0 then do not change directories.
 [ "${SHELL_WITH_GOPATH}" = "0" ] && exec /bin/sh "${@}"
 
+export OLD_GOPATH="${GOPATH}"
```
Comment (codenrhoden) [inline hack/shell-with-gopath.sh]: I don't see anything using this exported env var. Where is it used?

Comment (akutz) [reply hack/shell-with-gopath.sh]: Doh! That was something I added and then didn’t need. I’ll remove it. Good catch.

Comment (akutz) [reply hack/shell-with-gopath.sh]: Doh! That was something I added and then didn’t need. I’ll remove it. Good catch.

**Comment by akutz** [comment]
Can someone please lgtm this? @codenrhoden ? @figo ?

**Comment by codenrhoden** [comment]
yep.

/lgtm

**Pull request merged**